terra/compute/virtualenv.py

```diff
@@ -64,6 +65,20 @@ def run_service(self, service_info):
     executable = distutils.spawn.find_executable(service_info.command[0],
                                                  path=env['PATH'])
 
+    # Check if the executable was found in the virtualenv_dir.
+    # If it wasn't, warn the user in case they made a mistake
+    if is_subdir(executable, settings.compute.virtualenv_dir):
+      logger.warning("""
+                     Couldn't find command {} in virtualenv_dir {}. Using {}
+                     instead. If you meant to bypass the virtualenv dir, then
+                     feel free to ignore this message. If you weren't expecting
+                     this, then make sure the compute.virtualenv_dir is
+                     correct.
+                     """
```

Comment:
Why are you adding these spaces and newlines to this string?

Reply:
Because I'm writing out a multi-line string. How else would you recommend I format very long log messages?

Reply:
```
      logger.warning('Couldn't find command {} in virtualenv_dir {}. Using {} '
                     'instead. If you meant to bypass the virtualenv dir, then '
                     'feel free to ignore this message. If you weren't expecting '
                     'this, then make sure the compute.virtualenv_dir is '
                     'correct.')
```

Reply:
You have two choices for the formatting, use f-strings like you are _supposed_ to, or put an extra `()` around the multiline string and .format that, f that!

Reply:
Blammo!